File: extension/shared/lexicon.js
export const LEXICON = {
  he: {
    negative_roots: [
      "כועס","כועסת","כעס","מתוסכל","מתוסכלת","מבאס","מעצבן","מרגיז","נמאס",
      "בעיה","בעייתי","בעייתית","חמור","חמורה","לא תקין","לא תקינה","לא סביר","בלתי סביר",
      "לא הגיוני","הזוי","מביך","מיותר","כשל","כישלון","ממש לא","ממש ממש","פשוט לא","אין מצב","אין סיכוי"
    ],
    pressure_phrases: [
      "עדיין לא","עדיין לא חזרת","עדיין לא החזרת תשובה","עדיין לא קיבלתי תשובה",
      "לא קיבלתי תשובה","לא ענית","לא ענית לי","למה לא ענית","כמה פעמים","שלחתי כבר",
      "אני מחכה כבר","מחפשת אותך","כל השבוע","כבר שבוע","דחוף","בדחיפות","מייד","עכשיו",
      "תבדקו אלף פעמים","תבדקו מיליון פעמים","אלף פעמים","מיליון פעמים","מצפה לתגובה","תגובה מיידית","זה חייב","חייב"
    ],
    accusation_phrases: [
      "את לא","אתה לא","אתם לא","את לא מגיבה","את מתעלמת","לא עשית","לא טיפלת",
      "הבטחת ולא","אמרת ולא","סיכמנו ולא","היית אמור","היית אמורה","זה עלייך","זה באשמתך","אשמתך"
    ],
    insult_phrases: [
      "אני שונאת אותך","שונאת אותך","שונא אותך","תתבייש","תתבישי","תתבישו",
      "מטופש","טיפשי","אידיוט","אידיוטי","מגוחך","פתטי","חוסר אחריות","חוסר מקצועיות","זלזול","בושה","זאת בושה","ביזיון"
    ],
    softeners: ["בבקשה","תודה","תודה רבה","אשמח","אם אפשר","כשנוח לך","כשיתאפשר","בלי לחץ","ללא לחץ","מנסה להבין","רק לוודא","אשמח לעדכון"],
    escalation_signs: ["!!","???","?!","!?","!!!!!","……","...!!!"]
  },
  en: {
    negative_roots: ["angry","upset","frustrated","frustrating","annoying","problem","issue","unacceptable","ridiculous","unreasonable","not ok","not okay","this isn't ok","this is not ok"],
    pressure_phrases: ["still no answer","still waiting","asap","urgent","immediately","why didn't you","you didn't reply","you never replied","this must","you have to","no choice"],
    accusation_phrases: ["you didn't","you never","you ignored","you forgot","you failed","you were supposed to","it's your fault"],
    insult_phrases: ["i hate you","hate you","shame on you","stupid","idiotic","nonsense","pathetic","unprofessional"],
    softeners: ["please","thanks","thank you","whenever you can","no rush"],
    escalation_signs: ["!!","???","?!","!?","!!!!!","……","...!!!"]
  }
};

export function quickRiskScoreClient(text, lang = "he") {
  if (!text || !text.trim()) return 0;
  const lex = LEXICON[lang] || LEXICON.he;
  const raw = text.trim();
  const t = raw.toLowerCase();
  let score = 0;

  if (lex.escalation_signs.some(m => raw.includes(m))) score += 2;
  if (lex.insult_phrases.some(p => t.includes(p.toLowerCase()))) score += 8;

  lex.negative_roots.forEach(w => { if (t.includes(w.toLowerCase())) score += 2; });
  lex.pressure_phrases.forEach(w => { if (t.includes(w.toLowerCase())) score += 3; });
  lex.accusation_phrases.forEach(w => { if (t.includes(w.toLowerCase())) score += 3; });

  lex.softeners.forEach(w => { if (t.includes(w.toLowerCase())) score -= 1; });

  return Math.max(score, 0);
}

File: backend/app/ai/lexicon.py
# -*- coding: utf-8 -*-

ESCALATION_SIGNS = ["!!!!", "???", "?!", "!?", "!!!!!", "……", "...!!!"]

HE = {
    "negative_roots": [
        # כעס ותסכול
        "כועס", "כועסת", "כעס",
        "מתוסכל", "תסכול",
        "עצבני", "עצבנות",
        "מעצבן", "מרגיז",
        "נמאס", "נמאס לי",

        # בעייתיות ושיפוט
        "בעיה", "בעייתי", "בעייתית",
        "חמור", "חמורה",
        "לא תקין", "לא תקינה",
        "לא סביר", "בלתי סביר",
        "לא הגיוני", "הזוי",
        "מביך", "מיותר",
        "כשל", "כישלון",

        # ביטויי שלילה חזקים
        "ממש לא", "ממש ממש",
        "פשוט לא",
        "אין מצב", "אין סיכוי",
        "לא מקובל", "לא acceptable",
    ],

    "pressure_phrases": [
        # דחיפות ולחץ
        "עדיין לא", "עוד לא",
        "לא חזרת", "לא חזרת אליי",
        "לא קיבלתי תשובה", "לא ענית",
        "למה לא ענית", "למה אין תשובה",
        "כמה פעמים",
        "שלחתי כבר",
        "אני מחכה", "מחכה כבר",
        "כבר שבוע", "כבר ימים",

        "דחוף", "בדחיפות",
        "מייד", "מיידית", "עכשיו",
        "זה חייב", "חייב",
        "מצפה לתגובה", "תגובה מיידית",

        # הגזמה
        "אלף פעמים", "מיליון פעמים",
    ],

    "accusation_phrases": [
        # האשמה ישירה
        "את לא", "אתה לא", "אתם לא",
        "את לא מגיבה", "אתה לא מגיב",
        "את מתעלמת", "אתה מתעלם",

        "לא עשית", "לא טיפלת",
        "לא בדקת", "לא עדכנת",
        "הבטחת ולא", "אמרת ולא",
        "סיכמנו ולא",

        "היית אמור", "היית אמורה",
        "זה עלייך", "זה עליך",
        "באשמתך", "אשמתך",
    ],

    "insult_phrases": [
        # עלבונות וזלזול
        "שונא אותך", "שונאת אותך",
        "תתבייש", "תתבישי",
        "מטופש", "טיפשי",
        "אידיוט", "אידיוטי",
        "מגוחך", "פתטי",

        "חוסר אחריות",
        "חוסר מקצועיות",
        "זלזול",
        "בושה", "ביזיון",
    ],

    "explicit_emotion_phrases": [
        # התפרצות רגשית
        "די כבר", "די נו", "די!",
        "נמאס לי",
        "אין לי סבלנות",
        "כמה אפשר",
        "נו באמת",

        "אני כועסת", "אני כועס",
        "אין לי כוח",
        "תעני כבר",
        "למה את לא עונה",
    ],

    "softeners": [
        # מרככים (דווקא חשוב להשאיר)
        "בבקשה", "תודה", "תודה רבה",
        "אשמח", "אם אפשר",
        "כשנוח לך", "כשיתאפשר",
        "בלי לחץ", "ללא לחץ",
        "רק לבדוק", "רק לוודא",
        "אשמח לעדכון",
    ],

    "escalation_signs": ESCALATION_SIGNS,
}

EN = {
    "negative_roots": [
        "angry", "anger",
        "upset",
        "frustrat",   # frustrated / frustrating
        "annoy",
        "problem", "issue",
        "unacceptable",
        "ridiculous",
        "unreasonable",
        "not ok", "not okay",
        "this isn't ok",
    ],

    "pressure_phrases": [
        "still no answer",
        "still waiting",
        "asap", "urgent",
        "immediately", "right now",
        "why didn't you",
        "you didn't reply",
        "you never replied",
        "this must",
        "you have to",
        "no choice",
    ],

    "accusation_phrases": [
        "you didn't",
        "you never",
        "you ignored",
        "you forgot",
        "you failed",
        "you were supposed to",
        "it's your fault",
    ],

    "insult_phrases": [
        "i hate you", "hate you",
        "shame on you",
        "stupid", "idiotic",
        "pathetic",
        "unprofessional",
    ],

    "explicit_emotion_phrases": [
        "enough already",
        "i'm angry", "i am angry",
        "i'm upset",
        "fed up",
        "what's going on",
        "why aren't you replying",
        "answer already",
    ],

    "softeners": [
        "please", "thanks", "thank you",
        "whenever you can",
        "no rush",
        "just checking",
    ],

    "escalation_signs": ESCALATION_SIGNS,
}


LEXICON = {"he": HE, "en": EN}
